src/redux/greetings/greetingSlice.js

```diff
@@ -0,0 +1,40 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+const baseUrl = 'http://127.0.0.1:3000/api/greetings/random_greeting';
+
+const initialState = {
+  greetings: [],
+  isLoading: false,
+  error: null,
+};
+
+export const fetchGreetings = createAsyncThunk(
+  'greetings/fetchGreetings',
+  async () => {
+    const response = await fetch(baseUrl);
+    const data = await response.json();
+    return data;
+  },
+);
+
+const greetingSlice = createSlice({
+  name: 'greetings',
+  initialState,
+  reducers: {},
+  extraReducers(builder) {
+    builder
+      // .addCase(fetchGreetings.pending, (state) => {
+      //   state.isLoading = true;
+      // })
+      .addCase(fetchGreetings.fulfilled, (state, action) => {
+        state.greetings = action.payload;
+        state.isLoading = false;
+      });
+    // .addCase(fetchGreetings.rejected, (state, action) => {
+    //   state.error = action.error.message;
+    //   state.isLoading = false;
+    // });
```

Comment:
- [ ] kindly consider removing out all the commented codes to make your code base look clean cause niw they are making it lokk messy